src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2987,6 +2988,51 @@ private void QueryInternalPreparation(IndexQueryServerSide query)
                             {
                                 result.Results = reader.FacetedQuery(facetQuery, queryContext.Documents, GetOrAddSpatialField, token.Token);
                                 result.TotalResults = result.Results.Count;
+
+                                if (query.Metadata.Includes?.Length > 0)
+                                {
+                                    var cmd = new IncludeDocumentsCommand(this.DocumentDatabase.DocumentsStorage, queryContext.Documents, query.Metadata.Includes, true);
+                                    foreach (string include in query.Metadata.Includes)
```

Comment:
I don't want to do that, because we are not using `BlittableJsonObject` here, we are using the facet value, and I'm not sure if we want to introduce that behavior that deeply here.